poly-synth/src/poly-synth.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class PolySynth {
+  /**
+   * @class  PolySynth
+   * @constructor
+   * PolySynth manages the life cycle of voices.
+   * @param {AudioContext} context the audio context
+   */
+  constructor(context) {
+    if (!(context instanceof AudioContext))
+      throw context + ' is not a valid audio context.';
+
+    this.context_ = context;
+    this.sustainedVoices_ = [];
+    this.releasedVoices_ = [];
+
+    // The default maximum cutoff is set near the upper edge of human hearing
+    // and the minimum cutoff is near the lower edge of human hearing.
+    this.lowPassMaxCutoff = 12000;
+    this.lowPassMinCutoff = 60;
+
+    // By default, the filter should not affect audible frequencies.
+    this.lowPassCutoff_ = this.lowPassMaxCutoff;
+    this.output = new GainNode(this.context_);
+  }
+
+  /**
+   * Create a new voice and add it to the map of active voices.
+   * @param {String} note the note to be played, e.g. A4 for an octave four A
+   * @param {Number} frequency the corresponding frequency of the note, e.g 440
+   */
+  playNote(note, frequency) {
+    let voice = new PolySynthVoice(
```

Comment:
As I pointed out, this call can be reduced to `new PolySynthVoice(this)` if you can refactor the code properly.